monix-reactive/shared/src/test/scala/monix/reactive/internal/operators/GroupBySuite.scala

```diff
@@ -108,6 +109,23 @@ object GroupBySuite extends BaseOperatorSuite {
     assertEquals(fallbackTick, 3)
   }
 
+  test("on error groups should also error") { implicit s =>
+    var groupsCompleted = 0
+
+    val observable =
+      Observable(1, 2, 3)
+        .mapEval {
+          case n if n < 3 => Task.pure(n)
+          case _ => Task.raiseError(new RuntimeException)
+        }
+        .groupBy(identity)
+        .mapEval(_.completedL >> Task(groupsCompleted += 1))
+        .runAsyncGetLast
+
+    s.tick()
+    assertEquals(groupsCompleted, 0)
```

Comment:
Before the changes in this pull request, this test fails with `groupsCompleted = 2`.

Reply:
I think we should check for error instead of onComplete

Reply:
Agreed, fixed in https://github.com/monix/monix/pull/1190/commits/25b2dd6e8bee4bd995ff3efc3af939af98121105.